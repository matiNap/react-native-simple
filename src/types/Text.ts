import { SimpleFontSize, SimpleTextColor } from '.';
import { TextStyle, StyleProp, TextProperties } from 'react-native';

export interface LocalTextProps {
    fontSize?: SimpleFontSize;
    color?: SimpleTextColor;
    style?: StyleProp<TextStyle>;
}
export default interface Props extends TextProperties, LocalTextProps {
    children: string;
}
