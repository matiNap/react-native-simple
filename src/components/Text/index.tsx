import React from 'react';
import { Text, StyleProp, TextStyle } from 'react-native';
import useTheme from '../ThemeProvider/useTheme';
import { getThemeNumber, getThemeColor } from '../../helpers';
import { SimpleFontSize, SimpleTextColor } from '../../types';

interface Props {
    children: string;
    fontSize?: SimpleFontSize;
    color?: SimpleTextColor;
    style?: StyleProp<TextStyle>;
}

export default ({ children, style, ...props }: Props) => {
    const {
        typography: { fontSize },
        currentPalette,
    } = useTheme();
    const currentFontSize = getThemeNumber(
        props.fontSize,
        fontSize,
        fontSize.normal,
    );
    const textColor = getThemeColor(
        props.color,
        currentPalette.text,
        currentPalette.text.primary,
    );

    return (
        <Text style={[{ fontSize: currentFontSize, color: textColor }, style]}>
            {children}
        </Text>
    );
};
