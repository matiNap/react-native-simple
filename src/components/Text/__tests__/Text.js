import React from 'react';
import { shallow } from 'enzyme';
import Text from '../index';
import toJson from 'enzyme-to-json';

describe('Text component', () => {
    it('Should render', () => {
        const wrapper = shallow(<Text>text</Text>);

        expect(wrapper.length).toBe(1);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render text color', () => {
        const wrapper = shallow(<Text color="primary">text</Text>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render text fontSize', () => {
        const wrapper = shallow(<Text fontSize="medium">text</Text>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render with style', () => {
        const wrapper = shallow(
            <Text style={{ backgroundColor: 'blue' }}>text</Text>,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
