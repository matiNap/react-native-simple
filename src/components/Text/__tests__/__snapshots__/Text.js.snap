// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Should render 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontSize": undefined,
      },
      undefined,
    ]
  }
>
  text
</Text>
`;

exports[`Text component Should render text color 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontSize": undefined,
      },
      undefined,
    ]
  }
>
  text
</Text>
`;

exports[`Text component Should render text fontSize 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontSize": 25,
      },
      undefined,
    ]
  }
>
  text
</Text>
`;

exports[`Text component Should render with style 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontSize": undefined,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  text
</Text>
`;
